src/rdkafka_cgrp.c

```diff
@@ -2434,6 +2435,280 @@ static void rd_kafka_cgrp_offset_commit_tmr_cb (rd_kafka_timers_t *rkts,
 }
 
 
+/**
+ * @brief Incrementally add to an existing partition assignment
+ *        May update \p partitions but will not hold on to it.
+ *
+ * @returns an error object or NULL on success.
+ */
+static rd_kafka_error_t *
+rd_kafka_cgrp_incremental_assign (rd_kafka_cgrp_t *rkcg,
+                                  rd_kafka_topic_partition_list_t
+                                  *partitions) {
+        int i;
+        rd_kafka_resp_err_t err;
+
+        rd_assert(partitions);
+
+        if (rd_kafka_fatal_error_code(rkcg->rkcg_rk)) {
+                rd_kafka_dbg(rkcg->rkcg_rk, CGRP|RD_KAFKA_DBG_CONSUMER,
+                             "ASSIGN", "Group \"%s\": consumer is in "
+                             "a failed state, treating incremental "
+                             "assign of %d partitions as unassign of all"
+                             "%d partitions in the current assignment",
+                             rkcg->rkcg_group_id->str, partitions->cnt,
+                             !rkcg->rkcg_assignment ? 0
+                             : rkcg->rkcg_assignment->cnt);
+                err = rd_kafka_cgrp_assign(rkcg, NULL);
+                rd_assert(err);
```

Comment:
it should always return an error because the consumer is in a failed state. i have now made the assert more specific to check that err is fatal. this purpose is just fail fast in the case of developer error, and it adds a bit to code clarity. but will do as you direct if you want it out.

Reply:
changing my mind, removing the assert because i think it's more likely you'll accept it.